### PR TITLE
React tables migration

### PR DESCRIPTION
**Pull request by AmbossKeegan**
- finished code-wide migration to Tanstack react-tables
- deleted old dependency & files
- updated package.json / package-lock.json

### CODE REVIEW

File: src/client/src/components/table/ColumnConfigurations.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import { Table } from '@tanstack/react-table';
```
Comment (secondl1ght) [inline src/client/src/components/table/ColumnConfigurations.tsx:1]: Was the main change here just moving this into it's own component file or did the functionality need to be changed with the new package? (didn't do a full review on this file)

File: src/client/src/components/table/DebouncedInput.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect } from 'react';
```
Comment (secondl1ght) [inline src/client/src/components/table/DebouncedInput.tsx:1]: Was the main change here just moving this into it's own component file or did the functionality need to be changed with the new package? (didn't do a full review on this file)

File: src/client/src/components/table/index.tsx [hunk 3]
```diff
@@ -1,312 +1,219 @@
...
+        :last-child {
+          td {
+            border-bottom: 0;
+          }
+        }
+      }
+      .cursor {
+        cursor: pointer;
+      }
+      ,
```
Comment (secondl1ght) [inline src/client/src/components/table/index.tsx:60]: Syntax error here - not needed `,` character.

**Review by secondl1ght — CHANGES_REQUESTED** [review]
Nice work, left some comments inline!

File: src/client/src/views/home/reports/mempool/index.tsx [hunk 4]
```diff
@@ -33,7 +49,7 @@ export const MempoolReport = () => {
     <CardWithTitle>
       <SubTitle>Mempool Fees</SubTitle>
       <Card>
-        <Table alignCenter={true} tableColumns={columns} tableData={data} />
+        <Table alignCenter={true} columns={columns} data={data} />
```
[the remark this answers is not in the document]
Comment (AmbossKeegan) [reply src/client/src/views/home/reports/mempool/index.tsx:52]: This table only ever has a single row in it. Its the low, medium, high fee table, so this one doesn't need sorting.

File: src/client/src/components/table/DebouncedInput.tsx (hunk 2 above)
Comment (AmbossKeegan) [reply src/client/src/components/table/DebouncedInput.tsx:1]: The main change was to put this in its own file to reduce file size in the file it was declared in, as well as being able to use this component elsewhere.

File: src/client/src/components/table/ColumnConfigurations.tsx (hunk 1 above)
Comment (AmbossKeegan) [reply src/client/src/components/table/ColumnConfigurations.tsx:1]: The main change was to put this in its own file to reduce file size in the file it was declared in, as well as being able to use this component elsewhere.

File: src/client/src/views/chain/utxos/ChainUtxos.tsx [hunk 5]
```diff
@@ -29,43 +29,46 @@ export const ChainUtxos = () => {
...
               <div style={{ whiteSpace: 'nowrap' }}>
                 {getAddressLink(row.original.address)}
               </div>
             ),
           },
           {
-            Header: 'Format',
-            accessor: 'address_format',
+            header: 'Format',
+            enableSorting: false,
```
[the remark this answers is not in the document]
Comment (AmbossKeegan) [reply src/client/src/views/chain/utxos/ChainUtxos.tsx:70]: I made some discretionary changes as to which columns ought to be sortable and not. To me, the addresses as well as their formats are not within the realm of things that people would sort by. I'm happy to revert and make all columns that we're sortable previously sortable again. However, I believe in the previous implementation everything was sortable by default, whereas now I am exercising a little more critical thinking around which columns makes sense to be able to sort.

**Review by secondl1ght — APPROVED** [review]

File: src/client/src/views/dashboard/widgets/external/mempool.tsx [hunk 6]
```diff
@@ -17,10 +17,10 @@ export const MempoolWidget = () => {
   }
 
   const columns = [
-    { Header: 'Fastest', accessor: 'fast' },
-    { Header: 'Half Hour', accessor: 'halfHour' },
-    { Header: 'Hour', accessor: 'hour' },
-    { Header: 'Minimum', accessor: 'minimum' },
+    { header: 'Fastest', accessorKey: 'fast' },
```
[the remark this answers is not in the document]
Comment (SazKeegan) [reply src/client/src/views/dashboard/widgets/external/mempool.tsx:20]: The mempool table looks alright on mobile. Desktop and mobile there is no wrap.

[the remark this answers is not in the document]
Comment (AmbossKeegan) [reply src/client/src/views/dashboard/widgets/external/mempool.tsx:20]: The Mempool table looks alright on mobile and desktop. There is no wrap on either.

File: src/client/src/views/dashboard/widgets/external/mempool.tsx [hunk 7]
```diff
@@ -17,10 +17,10 @@ export const MempoolWidget = () => {
...
+    { header: 'Half Hour', accessorKey: 'halfHour' },
+    { header: 'Hour', accessorKey: 'hour' },
```
[the remark this answers is not in the document]
Comment (AmbossKeegan) [reply src/client/src/views/dashboard/widgets/external/mempool.tsx:22]: The Mempool table looks alright on mobile and desktop. There is no wrap on either.

**Pull request merged**